src/mon/Monitor.cc

```diff
@@ -2863,6 +2871,15 @@ void Monitor::log_health(
          << p.first << ")";
       clog->health(p.second.severity) << ss.str();
 
+      Formatter *f = Formatter::create("json-pretty");
```

Comment:
We should log even for health check update so this should be executed when health check is logged.

Reply:
to avoid generating too much output I'd suggest making the interval for logging details longer, e.g. not logging it if we have done so in the past 5 minutes.

The output can be pretty large in an unhealthy cluster.

Reply:
@jdurgin Logging `ceph health detail` on health check failure would also be useful. We do have checks to skip logging same info if there is no change in health status or summary. This will give us sequence of events leading to WARN/ERR state.